spec/system/placements/schools/placements/secondary_school_user_edits_a_placement_spec.rb

```diff
@@ -0,0 +1,404 @@
+  end
+
+  def and_i_see_a_term_updated_success_message
+    expect(page).to have_success_banner("Expected date updated")
+  end
+
+  def when_i_click_on_select_a_mentor
+    click_on "Select a mentor"
+  end
+
+  def then_i_see_the_select_a_mentor_page
+    expect(page).to have_title("Select a mentor - Placement details - Manage school placements - GOV.UK")
+    expect(primary_navigation).to have_current_item("Placements")
+    expect(page).to have_element(:span, text: "Placement details", class: "govuk-caption-l")
+    expect(page).to have_element(:legend, text: "Select a mentor", class: "govuk-fieldset__legend")
+    expect(page).to have_element(:div, text: "Some placements may have more than one mentor. For example, if a mentor works part time.", class: "govuk-hint")
+    expect(page).to have_element(:span, text: "My mentor is not listed", class: "govuk-details__summary-text")
+    expect(page).to have_link("Cancel", href: "/schools/#{@hogwarts_school.id}/placements/#{@placement.id}")
+  end
+
+  def when_i_click_on_my_mentor_is_not_listed
+    # click_on does not work for this element
+    find("span.govuk-details__summary-text", text: "My mentor is not listed").click
+  end
+
+  def then_i_see_the_mentor_not_listed_details_text
+    expect(page).to have_element(:div, text: "You must add mentors to your school's profile before they can be assigned to a specific placement.", class: "govuk-details__text")
+    expect(page).to have_link("add mentors to your school's profile", href: "/schools/#{@hogwarts_school.id}/mentors/new")
+  end
+
+  def and_i_see_my_mentors
+    expect(page).to have_field("John Smith", type: :checkbox)
+    expect(page).to have_field("Jane Doe", type: :checkbox)
+    expect(page).to have_field("Not yet known", type: :checkbox)
+  end
+
+  def and_i_select_john_smith
+    check "John Smith"
+  end
+
+  def then_i_see_the_placement_details_page_with_john_smith
+    expect(page).to have_title("English - Manage school placements - GOV.UK")
+    expect(primary_navigation).to have_current_item("Placements")
+    expect(page).to have_tag("Available", "turquoise")
+    expect(page).to have_summary_list_row("Subject", "English")
+    expect(page).to have_summary_list_row("Academic year", "This year (#{@current_academic_year_name})")
+    expect(page).to have_summary_list_row("Expected date", "Autumn term, Summer term")
+    expect(page).to have_summary_list_row("Mentor", "John Smith")
+    expect(page).to have_summary_list_row("Provider", "Assign a provider")
+    expect(page).to have_element(:div, text: "You can preview this placement as it appears to providers.", class: "govuk-inset-text")
+  end
+
+  def and_i_see_a_mentor_updated_success_message
+    expect(page).to have_success_banner("Mentor updated")
+  end
+
+  def when_i_click_on_change_mentor
+    click_on "Change Mentor"
+  end
+
+  def when_i_select_jane_doe
```

Comment:
This was unclear in the scenario that you were unselecting `John Smith`. I would suggest doing this as a separate step